libraries/hotstuff/qc_chain.cpp

```diff
@@ -589,7 +591,7 @@ namespace eosio { namespace hotstuff {
       if (_log) ilog(" === ${id} process_new_view === ${qc}", ("qc", msg.high_qc)("id", _id));
 #endif
       if (update_high_qc(msg.high_qc)) {
-         ++_state_version;
+         ++_state_version; // should be OK; update_high_qc() should not throw
```

Comment:
Maybe OK now, but if someone adds an EOS_ASSERT in there all bets are off. 
Better to add: `auto increment_version = fc::make_scoped_exit([this]() { ++_state_version; });` **before** calling `update_high_qc` and calling `increment_version.cancel()` if the function returns false.